db/database.go

```diff
@@ -24,38 +25,110 @@ type Database interface {
 	Transact(fn TxFunc, opts ...TxOpt) (interface{}, error)
 	Metadata() (map[string]string, error)
 	Stats() sql.DBStats
+	WithContext(ctx context.Context) Database
+}
+
+type QueryOptions struct {
+	logger logging.Logger
+	ctx    context.Context
 }
 
 type SqlxDatabase struct {
-	db *sqlx.DB
+	db           *sqlx.DB
+	queryOptions *QueryOptions
 }
 
 func NewSqlxDatabase(db *sqlx.DB) *SqlxDatabase {
 	return &SqlxDatabase{db: db}
 }
 
+func (d *SqlxDatabase) getLogger() logging.Logger {
+	if d.queryOptions != nil {
+		return d.queryOptions.logger
+	}
+	return logging.Default()
+}
+
+func (d *SqlxDatabase) getContext() context.Context {
+	if d.queryOptions != nil {
+		return d.queryOptions.ctx
+	}
+	return context.Background()
+}
+
+func (d *SqlxDatabase) WithContext(ctx context.Context) Database {
+	return &SqlxDatabase{
+		db: d.db,
+		queryOptions: &QueryOptions{
+			logger: logging.Default().WithContext(ctx),
+			ctx:    ctx,
+		},
+	}
+}
+
 func (d *SqlxDatabase) Close() error {
 	return d.db.Close()
 }
 
+// reportFinish performs fn and logs a "done" report if its duration was long enough.
```

Comment:
update the code doc - needs to start with the function name - performAndReport

Reply:
Yeah, thanks!